### PR TITLE
Pass runfiles paths more robustly

### PR DESCRIPTION
**Pull request by comius**
Pass jvm-abi-gen and kotlin-compiler with system property

This passes runfiles path returned from rootpath function, which is correct in setups with and without using bzlmod.

Use Dep.fromLabel in tests to find dependencies 

The `Dep.fromLabel` works well in combination with `kt_rules_test`. The same
mangling and demangling is used to find runfiles.

This is a slight simplification, which as well prepares the repo for
bzlmod.

In addition workaround for autovalue is replaced with a filegroup trick,
which works with newer rules_jvm_external repository (where visibility
of .jar targets is limited).

### CODE REVIEW

File: src/test/kotlin/io/bazel/kotlin/defs.bzl [hunk 1]
```diff
@@ -35,6 +35,7 @@ def kt_rules_test(name, **kwargs):
         "//src/main/kotlin:jdeps-gen",
         "@com_github_jetbrains_kotlin//:annotations",
         "@com_github_jetbrains_kotlin//:jvm-abi-gen",
+        "@com_github_jetbrains_kotlin//:kotlin-compiler",
         "@com_github_jetbrains_kotlin//:kotlin-stdlib",
         "@com_github_jetbrains_kotlin//:kotlin-stdlib-jdk7",
         "@com_github_jetbrains_kotlin//:kotlin-stdlib-jdk8",
```
Comment (comius) [inline src/test/kotlin/io/bazel/kotlin/defs.bzl:41]: cc @fmeum line 45 is where `rootpath` is used

File: src/main/starlark/core/repositories/bzlmod_setup.bzl [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    for segment in path.split("/"):
+        root = root.get_child(segment)
+    repository_ctx.symlink(root, ".")
+
+_local_repository = repository_rule(
+    implementation = _local_repository_impl,
+    attrs = {
+        "path": attr.string(mandatory = True),
+    },
+)
```
Comment (comius) [inline src/main/starlark/core/repositories/bzlmod_setup.bzl:63]: cc @meteorcloudy @Wyverald 
Is there a better way to do local repos with bzlmod?

Comment (Wyverald) [reply src/main/starlark/core/repositories/bzlmod_setup.bzl:63]: unfortunately, still not -- https://github.com/bazelbuild/bazel/issues/15412 is surprisingly hard to fix, involving a rather large refactor.

Comment (comius) [reply src/main/starlark/core/repositories/bzlmod_setup.bzl:60]: Uf, this is also a blocker for Android, with android_sdk_repository being native.

**Comment by restingbull** [comment]
Please split the target refactorings from the bzlmod changes -- it'll be easier to keep track of what moves where.

File: src/main/starlark/core/repositories/bzlmod_setup.bzl (hunk 2 above)
Comment (meteorcloudy) [reply src/main/starlark/core/repositories/bzlmod_setup.bzl:60]: How hard is it to starlarkfiy andorid_sdk_repository rule? /cc @ahumesky @ted-xie

**Pull request merged**